### DISCUSSION TITLE
How to improve the CodeMeta mappings of the crosswalk table

### DISCUSSION THREAD

**Discussion by moranegg** [codemeta/codemeta#336] [Ideas]
I would like to start a discussion on the topic of improving the crosswalks/mappings in:

## Goals
- Have machine actionable mappings to be easily integrated into translation tools
- Keep mappings that are FAIR=> Findable, Accessible, Interoperable and Reusable
- Keep track of the information about the mapping creation - metametadata

## Possible formats
Recently I've been introduced to the SSSOM format: https://mapping-commons.github.io/sssom/
which is  adapted to achieve the goals above.

## Risks
- harder to get contributions
- needs to be compatible with the table reconstruction scripts / or low effort to change

Please comment this discussion with your take on this potential enhancement of the CodeMeta mappings.

Reference:
- https://github.com/codemeta/codemeta/issues/265

**Comment by dgarijo** [comment]
Maybe we can do a smooth transition: first add a mechanism to introduce metadata (who did it, a column for mapping rationale) and then the means to state whether something is related, broader, narrower than a target concept.

**Comment by nsjuty** [comment]
The mappings survey from Yann for t4.4 in FAIR-IMPACT captures some of this information - the results could be useful. Also aligns with the framework we will present in terms of metametadata, in terms of mapping creation context. 
Links:
https://fair-impact.eu/news/collecting-ways-doing-mappings-take-survey
https://fair-impact.eu/events/fair-impact-events/developing-mapping-process-framework

I'm not quite ready with the framework 'information capture' on use, but I'm trying to make it low barrier (drop down options etc).

**Comment by moranegg** [comment]
@nsjuty do you have a template that we could use?

If we want to work on this in the repository, we can create a new folder called `FAIR-crosswalks` or `FAIR-mappings` where we can start creating this improved / FAIRer mappings.

Another option is to do this directly in a mapping registry, but this might be harder to maintain.

Any thoughts on ways to move forward from PMC?

**Comment by moranegg** [comment]
## A proposed methodology (thanks UPM!! @esgg and @dgarijo )
1. The mappings will be created in a separate folder in the [CodeMeta repository](https://github.com/codemeta/codemeta), not in the current "crosswalks" folder. 
2. For each mapping, we will follow the following steps:
  A. Create YAML "Meta"Metadata File – this file contains the information about the mapping and not the mapping itself.
  B. Prepare CSV Mapping File – Structure mapping data in a CSV with columns for source term, target term, and relation type (e.g., same_as, part_of). This is the "simple" mapping in plain .csv like before with a few improvements.
  C. Generate SSSOM File – Run generate_crosswalk.py to convert the CSV into an .sssom.tsv format for interoperability.
3. Each mapping will have a w3id and this id will be visible in the YAML file.

Full example here: https://github.com/oeg-upm/codemeta-crosswalk-mapping-methodology/tree/main/examples

Please comment here.

**Comment by moranegg** [comment]
Hello CodeMeta community,

During the SciCodes 2025 workshop, we have started to put in place the proposed methodology.
To help us keep track of the effort, we now have our first GitHub project: https://github.com/orgs/codemeta/projects/1

This discussion will be the location to give feedback and debate.
Specific pain points will be discussed in specific issues that are labeled `SSSOM Methodology`.

This work is parallel to an upcoming v4.0, which is tagged with the v4.0 milestone.
Reminder: v4.0 release includes changes to the vocabulary itself.

Thank you for your engagement!

Ping to the CodeMeta PMC for information:

@abbycabs @arfon  @cboettig @dgarijo @mbjones  @tmorrell  @progval
